internal/pkg/plugin/binary.go

```diff
@@ -162,10 +330,101 @@ func (m *Meta) configName() string {
 // DESCR[1]: Sifmanifest
 //   - Datatype: sif.DataGenericJSON
 func isPluginFile(fimg *sif.FileImage) bool {
-	return false
+	if len(fimg.DescrArr) < 2 {
+		return false
+	}
+
+	if !fimg.DescrArr[0].Used {
+		return false
+	}
+
+	if fimg.DescrArr[0].Datatype != sif.DataPartition {
+		return false
+	}
+
+	if fstype, err := fimg.DescrArr[0].GetFsType(); err != nil {
+		return false
+	} else if fstype != sif.FsRaw {
+		return false
+	}
+
+	if partype, err := fimg.DescrArr[0].GetPartType(); err != nil {
+		return false
+	} else if partype != sif.PartData {
+		return false
+	}
+
+	if !fimg.DescrArr[1].Used {
+		return false
+	}
+
+	if fimg.DescrArr[1].Datatype != sif.DataGenericJSON {
+		return false
+	}
+
+	return true
 }
 
 // getManifest will extract the Manifest data from the input FileImage
 func getManifest(fimg *sif.FileImage) pluginapi.Manifest {
-	return pluginapi.Manifest{}
+	var (
+		manifest pluginapi.Manifest
+		start    = fimg.DescrArr[1].Fileoff
+		end      = start + fimg.DescrArr[1].Filelen
+		data     = fimg.Filedata[start:end]
+	)
+
+	if err := json.Unmarshal(data, &manifest); err != nil {
+		fmt.Println(err)
+	}
+
+	return manifest
+}
+
+// List returns all the singularity plugins installed in libexecdir in
+// the form of a list of Meta information
+func List(libexecdir string) ([]*Meta, error) {
```

Comment:
I commented this somewhere else, but maybe we can return `map[string]*Meta`, unless we see a reason not to. Key can be the SHA name of the meta file, since it's guaranteed to be unique per name, we shouldn't have any hash map collisions. What're your thoughts here @mem ? I don't see any immediate drawbacks to using a map, can you think of any?